src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/CohereEmbeddingClient.java

```diff
@@ -30,23 +26,18 @@
  * Interface that accepts a list of texts that needs to be vectorized and returns embeddings based
  * of chosen Cohere model.
  */
-public class CohereEmbeddingClient implements EmbeddingProvider {
-  private EmbeddingProviderConfigStore.RequestProperties requestProperties;
-  private String modelName;
-  private String baseUrl;
+public class CohereEmbeddingClient extends EmbeddingProvider {
+  private static final String providerId = ProviderConstants.COHERE;
```

Comment:
As per my other notes, can be moved to super-class as field, passed by subtype constructor.
And then referenced like other `protected` fields.

Reply:
(as per earlier note, can't be done)